src/replication.c

```diff
@@ -1349,6 +1351,29 @@ void removeRDBUsedToSyncReplicas(void) {
     }
 }
 
+/* Close the repldbfd and reclaim the page cache if the client hold
+ * the last reference to replication DB */
+void closeRepldbfd(client *myself) {
+    listNode *ln;
+    listIter li;
+    int shouldInvalidateCache = 1;
+    listRewind(server.slaves,&li);
+    while((ln = listNext(&li))) {
+        client *slave = ln->value;
+        if (slave != myself && slave->replstate == SLAVE_STATE_SEND_BULK) {
+            shouldInvalidateCache = 0;
+            break;
+        }
+    }
+
+    if (shouldInvalidateCache) {
+        bioCreateCloseJob(myself->repldbfd, 0, 1);
```

Comment:
neat!
but again, maybe we can incrementally release the pages?
i guess that in this case it doesn't matter much since we had to accumulate the whole rdb file in the page cache anyway (unlike a persistence save, in which we can avoid that buildup), so it might not matter if we release it incrementally or in one go?